src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py

```diff
@@ -77,19 +80,37 @@ async def price_change(self, query_id: bytes, value_before: float) -> Optional[f
 
         Returns: float
         """
-        if query_id not in CATALOG_QUERY_IDS:
-            return None
-
-        query_tag = CATALOG_QUERY_IDS[query_id]
-
-        if query_tag in CATALOG_FEEDS:
-            datafeed: DataFeed[Any] = CATALOG_FEEDS[query_tag]
+        query_id = bytes(w3.keccak(query_data))
+        query_entry = query_from_query_catalog(qid=query_id.hex())
```

Comment:
Maybe `query_from_query_catalog` should return  a `Query` type or `None`, so you don't have to make this extra var and then later `query = query_entry.query`

Reply:
I'm using query_entry for the query and the tag, query_entry.query and query_entry.tag.